src/katgpucbf/fgpu/engine.py

```diff
@@ -1264,8 +1264,8 @@ async def _run_transmit(self, stream: "spead2.send.asyncio.AsyncStream") -> None
                 pass  # It's already logged by the chunk_finished callback
         stop_heap = spead2.send.Heap(send.FLAVOUR)
         stop_heap.add_end()
-        for substream_index in range(stream.num_substreams):
-            await stream.async_send_heap(stop_heap, substream_index=substream_index)
+        for substream_index in range(streams[0].num_substreams):
+            await streams[0].async_send_heap(stop_heap, substream_index=substream_index)
```

Comment:
It's likely that my spead2 fog is hindering my view, but why are we/you only sending the stop heap on one of the streams (instead of `each stream in streams`)?

Reply:
substream_index is still iterating over all the multicast groups. I *could* do something like `_multi_send` and partition the groups over the streams, but since it's just one heap I don't really care which network interface is used.

Reply:
I see, I see.